src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -169,6 +189,32 @@ internal static unsafe int strlen(byte* pString)
             return checked((int)(p - pString));
         }
 
+        internal unsafe static void* CoTaskMemAllocAndZeroMemory(global::System.IntPtr size)
+        {
+            void* ptr;
+
+            ptr = Interop.MemAlloc((UIntPtr)(void*)size).ToPointer();
+
+            // Interop.MemAlloc will throw OOMException if out of memory
+            System.Diagnostics.Debug.Assert(ptr != null);
+
+            // TODO: optimize using  memset
+            byte* pByte = (byte*)ptr;
+            long lSize = size.ToInt64();
+            while (lSize > 0)
+            {
+                lSize--;
+                (*pByte++) = 0;
+            }
+
+            return ptr;
+        }
+
+        internal unsafe static void CoTaskMemFree(void* p)
+        {
+            Interop.MemFree((IntPtr)p);
```

Comment:
Okay. In that case for allocation we should call into CoTaskMemAlloc instead of HeapAlloc in Windows. I will fix it.